chainer/functions/math/squared_difference.py

```diff
@@ -0,0 +1,40 @@
+from chainer import cuda
+from chainer import function
+from chainer.utils import type_check
+
+
+class SquaredDifference(function.Function):
+    """Squared difference of input variables."""
+
+    def check_type_forward(self, in_types):
+        type_check.expect(
+            in_types.size() == 2,
```

Comment:
Split this line from others because when in_types.size() == 1, in_types[1] causes index error.
```
type_check.expect(in_types.size() == 2)
type_check.expect(
    in_types[0].dtype.kind == 'f',
    ...
)
```